pkg/lint/rules.go

```diff
@@ -375,6 +375,24 @@ func ValidateDuplicateColumnNames(ctx context.Context, p *pipeline.Pipeline, ass
 	return issues, nil
 }
 
+func ValidateInvalidPythonModuleName(ctx context.Context, p *pipeline.Pipeline, asset *pipeline.Asset) ([]*Issue, error) {
+	var issues []*Issue
+
+	if asset.Type == "python" {
+		parentDirs := strings.Split(filepath.Dir(asset.DefinitionFile.Path), "/")
+		lastDir := parentDirs[len(parentDirs)-1]
+
+		// Check if the last directory contains a hyphen and is not in the 'assets' directory
+		if strings.Contains(lastDir, "-") && lastDir != "assets" {
+			issues = append(issues, &Issue{
+				Task:        asset,
+				Description: fmt.Sprintf("Invalid Python module name '%s' for asset '%s'. Directory names cannot contain hyphens ('-') as they cannot be imported in Python.", lastDir, asset.Name),
+			})
+		}
```

Comment:
I don't think the issue is just with the last dir:
- bruin asets can be in any depth in the file tree, e.g. `/my-folder/assets/level1/level2/level3/asset.py`
- the `-` existence should be checked in any path within the file tree.

here are some examples:
- VALID: `/my-folder/assets/level1/level2/level3/asset.py`
- VALID: `some-high-folder/above/assets/level1/asset.py`
- INVALID: `/myfolder/assets/one-with-dash/level2/level3/asset.py`
- INVALID: `/myfolder/assets/no_dash/but-here-dash/level3/asset.py`
- INVALID: `/myfolder/assets/no_dash/no_dash_either/this-time-dash/asset.py`

Reply:
@karakanb is it correct that we need to validate all dir path after `/my-folder/assets/` ?